### PR TITLE
Check parent roomauth for modjoin

### PR DESCRIPTION
**Pull request by MathyFurret**
Context: https://www.smogon.com/forums/threads/allow-room-owners-to-join-invite-only-subrooms.3702238/

### CODE REVIEW

File: server/rooms.ts [hunk 1]
```diff
@@ -558,7 +558,7 @@ export abstract class BasicRoom {
 		if (!Users.Auth.isAuthLevel(modjoinSetting)) {
 			Monitor.error(`Invalid modjoin setting in ${this.roomid}: ${modjoinSetting}`);
 		}
-		return Users.globalAuth.atLeast(user, modjoinSetting);
+		return this.auth.atLeast(user, modjoinSetting);
```
Comment (AnnikaCodes) [inline server/rooms.ts:561]: Wouldn't this cause issues if a subroom were set to modjoin `&`, thus allowing subroom auth to join but not the parent room's owner?

Comment (AnnikaCodes) [reply server/rooms.ts:561]: Never mind - it looks like groupchats restrict modjoin to +/% so this fix is OK.

**Comment by AnnikaCodes** [comment]
Thanks!

**Comment by mia-pi-git** [comment]
This seems to have made it so admins can no longer join private rooms. See https://github.com/smogon/pokemon-showdown/commit/4a24d3ee1effa26f300837772576c2f2fc0a254b.

**Pull request merged**